### PR TITLE
feat: sniff google sheets csv paths

### PR DESCRIPTION
**Pull request by tychoish**
:shrug:

#2604

### CODE REVIEW

File: crates/datafusion_ext/src/planner/relation/mod.rs [hunk 1]
```diff
@@ -235,12 +235,18 @@ impl<'a, S: AsyncContextProvider> SqlQueryPlanner<'a, S> {
 /// Returns a reference to table func by inferring which function to use from a
 /// given path.
 fn infer_func_for_file(path: &str) -> Result<OwnedTableReference> {
-    let ext = Path::new(path)
-        .extension()
-        .ok_or_else(|| DataFusionError::Plan(format!("missing file extension: {path}")))?
-        .to_str()
-        .ok_or_else(|| DataFusionError::Plan(format!("strange file extension: {path}")))?
-        .to_lowercase();
+    let ext = if path.contains("docs.google.com/spreadsheet") && path.trim().ends_with("output=csv")
```
Comment (vrongmeal) [inline crates/datafusion_ext/src/planner/relation/mod.rs:238]: Should we instead pass the `path` as `&Url` and check if there is a query parameter output with value `csv`? `path` ending with `output=csv` feels wrong...

Comment (tychoish) [reply crates/datafusion_ext/src/planner/relation/mod.rs:238]: It does feel wrong. 

I also wonder if we can change the content (mime?) type header without specifying an argument?

I guess also we can do some amount of heroics if we get a URL that _doesn't_ have the argument, we could add it?

**Comment by tychoish** [comment]
I poked at this a bit and the issue is that `ObjectStoreUrl::parse` returns this error in [these functions](https://github.com/glaredb/glaredb/blob/tycho/google-sheets-hack/crates/datasources/src/common/url.rs#L164-L184) which we're hitting, and I think this means we can't use read_csv, on the other hand I think we can probably just write a dedicated function and call reqwest directly. 

This is definitely not going to be a hack and will take a middle/moderate amount of work

**Comment by tychoish** [comment]
closing, but will reuse your test spreadsheet @talagluck

**Pull request closed without merging**